Sources/Networking/NetworkRequestPerformer.swift

```diff
@@ -28,4 +28,13 @@ public protocol NetworkRequestPerformer {
     /// - Returns: Returns a publisher that can be subscribed to, that performs the given request with the given behaviors.
     @available(iOS 13.0, *)
     @discardableResult func send(_ request: any NetworkRequest, requestBehaviors: [RequestBehavior]) -> AnyPublisher<NetworkResponse, NetworkError>
+    
+    /// Performs the given request with the given behaviors returning a JSON-decoded object with async/await, or throwing an error if unsuccessful.
+    ///
+    /// - Parameters:
+    ///   - request: The request to perform.
+    ///   - requestBehaviors: The behaviors to apply to the given request.
+    ///   - decoder: The JSON decoder to use when decoding the data.
+    /// - Returns: A JSON-decoded object.
+    func send<ResponseType: Decodable>(_ request: any NetworkRequest, requestBehaviors: [RequestBehavior], decoder: JSONDecoder) async throws -> ResponseType
```

Comment:
This looks good, but we should follow the existing patterns for the declarations, extensions, and conformances already set.

- This protocol declaration should include a method that returns a `NetworkResponse`, free from any JSON-related work, like decoding into a specific model type. See the `send` method above as an example. It would look something like:
```swift
send(_ request: any NetworkRequest, requestBehaviors: [RequestBehavior]) async throws -> NetworkResponse
```
- You can then add a convenience method to `NetworkRequestPerformer+JSON` that wraps the call to the async/await version of `send` declared here. That method would have the same signature that you’ve used in `NetworkRequestPerformer+JSON.swift`, but its implementation would use the async/await implementation instead of the combine JSON-wrapped implementation it currently uses.
- Finally, in the `NetworkController` conformance of `NetworkRequestPerformer` in `NetworkController.swift`, the async/await `send` method declared here should be implemented, again, without consideration for JSON decoding, as that's done entirely in the wrapper within `NetworkRequestPerformer+JSON.swift`.

---

Alternatively, if that can’t be accomplished for any reason, we should remove this declaration from `NetworkRequestPerformer.swift`, and keep what you have in `NetworkRequestPerformer+JSON.swift`.

Reply:
To the best of my knowledge I tried to make the changes, would you have time this week to go over your comments, it would really help me to get a better understanding of what the expectation is when we introduce new methods and updates to the `Networking` framework?

Reply:
Yeah, we’re close on this one, so once we complete the review/merge it, I’m happy to discuss these things further. I’m adding additional notes in my next review comments, so hopefully that’ll help to clarify some of this up front. After that, happy to discuss the rest!